OctoTiger does not compile with HPX master and CUDA 11.5

## Expected Behavior

Should compile.

## Actual Behavior

```
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp: In function 'std::enable_if_t<is_client_v<Client>, std::vector<T> > hpx::components::make_clients(std::vector<hpx::future<hpx::naming::id_type> >&&)':
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp:116:78: error: invalid use of 'this' in non-member function
  116 |             result.push_back(Client(HPX_MOVE(id)));
      |                                                                              ^   
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp:116:78: error: invalid use of 'this' in non-member function
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp:116:84: error: template argument 1 is invalid
  116 |             result.push_back(Client(HPX_MOVE(id)));
      |                                                                                    ^
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp:116:87: error: expected '>' before '&&' token
  116 |             result.push_back(Client(HPX_MOVE(id)));
      |                                                                                       ^ 
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp:116:87: error: expected '(' before '&&' token
  116 |             result.push_back(Client(HPX_MOVE(id)));
      |                                                                                       ^ 
      |                                                                                       (
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp:116:89: error: expected identifier before '>' token
  116 |             result.push_back(Client(HPX_MOVE(id)));
      |                                                                                         ^
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp: In function 'std::enable_if_t<is_client_v<Client>, std::vector<T> > hpx::components::make_clients(std::vector<hpx::shared_future<hpx::naming::id_type> >&&)':
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp:144:78: error: invalid use of 'this' in non-member function
  144 |             result.push_back(Client(HPX_MOVE(id)));
      |                                                                              ^   
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp:144:78: error: invalid use of 'this' in non-member function
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp:144:84: error: template argument 1 is invalid
  144 |             result.push_back(Client(HPX_MOVE(id)));
      |                                                                                    ^
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp:144:87: error: expected '>' before '&&' token
  144 |             result.push_back(Client(HPX_MOVE(id)));
      |                                                                                       ^ 
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp:144:87: error: expected '(' before '&&' token
  144 |             result.push_back(Client(HPX_MOVE(id)));
      |                                                                                       ^ 
      |                                                                                       (
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp:144:89: error: expected identifier before '>' token
  144 |             result.push_back(Client(HPX_MOVE(id)));
      |                                                                                         ^
gmake[2]: *** [CMakeFiles/hydrolib.dir/build.make:132: CMakeFiles/hydrolib.dir/src/unitiger/hydro_impl/flux_cuda_kernel.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp: In function 'std::enable_if_t<is_client_v<Client>, std::vector<Future> > hpx::components::make_clients(std::vector<hpx::future<hpx::naming::id_type> >&&)':
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp:116:78: error: invalid use of 'this' in non-member function
  116 |             result.push_back(Client(HPX_MOVE(id)));
      |                                                                              ^   
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp:116:78: error: invalid use of 'this' in non-member function
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp:116:84: error: template argument 1 is invalid
  116 |             result.push_back(Client(HPX_MOVE(id)));
      |                                                                                    ^
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp:116:87: error: expected '>' before '&&' token
  116 |             result.push_back(Client(HPX_MOVE(id)));
      |                                                                                       ^ 
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp:116:87: error: expected '(' before '&&' token
  116 |             result.push_back(Client(HPX_MOVE(id)));
      |                                                                                       ^ 
      |                                                                                       (
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp:116:89: error: expected identifier before '>' token
  116 |             result.push_back(Client(HPX_MOVE(id)));
      |                                                                                         ^
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp: In function 'std::enable_if_t<is_client_v<Client>, std::vector<Future> > hpx::components::make_clients(std::vector<hpx::shared_future<hpx::naming::id_type> >&&)':
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp:144:78: error: invalid use of 'this' in non-member function
  144 |             result.push_back(Client(HPX_MOVE(id)));
      |                                                                              ^   
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp:144:78: error: invalid use of 'this' in non-member function
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp:144:84: error: template argument 1 is invalid
  144 |             result.push_back(Client(HPX_MOVE(id)));
      |                                                                                    ^
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp:144:87: error: expected '>' before '&&' token
  144 |             result.push_back(Client(HPX_MOVE(id)));
      |                                                                                       ^ 
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp:144:87: error: expected '(' before '&&' token
  144 |             result.push_back(Client(HPX_MOVE(id)));
      |                                                                                       ^ 
      |                                                                                       (
/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/hpx/include/hpx/components/make_client.hpp:144:89: error: expected identifier before '>' token
  144 |             result.push_back(Client(HPX_MOVE(id)));
      |                                                                                         ^
gmake[2]: *** [CMakeFiles/hydrolib.dir/build.make:90: CMakeFiles/hydrolib.dir/src/unitiger/hydro_impl/hydro_kernel_interface.cpp.o] Error 1
gmake[2]: Leaving directory '/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/octotiger/build-kokkos'
gmake[1]: *** [CMakeFiles/Makefile2:176: CMakeFiles/hydrolib.dir/all] Error 2
gmake[1]: Leaving directory '/pscratch/sd/d/diehlpk/OctoTigerBuildChain/build/octotiger/build-kokkos'
gmake: *** [Makefile:136: all] Error 2
```


https://pastebin.com/vyMAUEDD


## Steps to Reproduce the Problem

... Please be as specific as possible while describing how to reproduce your problem.

  1. Compile HPX master with CUDA 11.5


## Specifications

... Please describe your environment

  - HPX Version: master
  - Platform (compiler, OS): Perlmutter

--- a/libs/full/components/include/hpx/components/make_client.hpp
+++ b/libs/full/components/include/hpx/components/make_client.hpp
@@ -76,7 +76,8 @@ namespace hpx { namespace components {
         return result;
     }
 
-#if !defined(HPX_CUDA_VERSION) || (HPX_CUDA_VERSION > 1104)
+    // this is broken at least up until CUDA V11.5
+#if !defined(HPX_CUDA_VERSION)
     template <typename Client>
     inline std::enable_if_t<traits::is_client_v<Client>, std::vector<Client>>
     make_clients(std::vector<hpx::id_type>&& ids)
@@ -104,7 +105,8 @@ namespace hpx { namespace components {
         return result;
     }
 
-#if !defined(HPX_CUDA_VERSION) || (HPX_CUDA_VERSION > 1104)
+    // this is broken at least up until CUDA V11.5
+#if !defined(HPX_CUDA_VERSION)
     template <typename Client>
     inline std::enable_if_t<traits::is_client_v<Client>, std::vector<Client>>
     make_clients(std::vector<hpx::future<hpx::id_type>>&& ids)
@@ -132,7 +134,8 @@ namespace hpx { namespace components {
         return result;
     }
 
-#if !defined(HPX_CUDA_VERSION) || (HPX_CUDA_VERSION > 1104)
+    // this is broken at least up until CUDA V11.5
+#if !defined(HPX_CUDA_VERSION)
     template <typename Client>
     inline std::enable_if_t<traits::is_client_v<Client>, std::vector<Client>>
     make_clients(std::vector<hpx::shared_future<hpx::id_type>>&& ids)

